### PR TITLE
Update `EvmLoader` to generate Yul code instead of bytecode

### PR DESCRIPTION
**Pull request by DoHoonKim8**
I worked on updating `EvmLoader` to generate Yul code instead of EVM bytecode. Also, I updated `kzg::decider::Decider` and `transcript::evm::EvmTranscript` to generate Yul.

The motivation of my work is to make EVM full plonk verifier to be auditable and more human readable. It's my first time to contribute to this project, so it will be really thankful to give me many comments if something is wrong.

### CODE REVIEW

**Comment by DoHoonKim8** [comment]
> failures:
        system::halo2::test::kzg::evm::test_plonk_zk_main_gate_with_range_with_mock_kzg_accumulator
        system::halo2::test::kzg::evm::test_plonk_zk_standard_plonk_rand
        system::halo2::test::kzg::evm::test_shplonk_zk_main_gate_with_range_with_mock_kzg_accumulator
        system::halo2::test::kzg::evm::test_shplonk_zk_standard_plonk_rand

I think some tests are failing because `solc` binary is not installed in CI server. (Currently `EvmLoader::compile` function assumes that `solc` binary is installed. [here](https://github.com/DoHoonKim8/plonk-verifier/blob/47092ce0990ffbd01a34aacb75055245dab1f963/src/loader/evm/loader.rs#L107))

I considered the following options:
   - searched for Solidity compiler written in Rust, but I could not find one.
   - add `solc` binary to the package which seems to be not a good idea
   - maybe server can install `solc` before running the test, however I don't know how to.

Is there any good way? 🙏 @han0110

**Comment by han0110** [comment]
Hey @DoHoonKim8 the CI is now installed with `solc@0.8.17` after #16, after rebase it should work for this PR.

**Review by han0110 — APPROVED** [review]
Looks good to me! We now can generate Yul for people to more easily look where to optimize. Thanks for the contribution!

Just added a small nitpick, after it's resolved we can merge this.

File: src/loader/evm/loader.rs [hunk 1]
```diff
@@ -525,22 +436,11 @@ impl EvmLoader {
...
-            .push(9)
-            .gas()
-            .swap(2)
-            .sub()
-            .sub()
-            .push(0)
-            .push(0)
-            .log1();
-    }
+    fn end_gas_metering(self: &Rc<Self>) {}
```
[the remark this answers is not in the document]
Comment (DoHoonKim8) [reply src/loader/evm/loader.rs:443]: Oh... I missed this function 😓 Thanks for implementing this on my behalf!

**Comment by DoHoonKim8** [comment]
> Looks good to me! We now can generate Yul for people to more easily look where to optimize. Thanks for the contribution!
> 
> Just added a small nitpick, after it's resolved we can merge this.

Thanks Han!! Your guide was really helpful for my contribution. Also I checked the [issue](https://github.com/privacy-scaling-explorations/plonk-verifier/issues/17), and I wish we can generate more memory-optimized and readable Yul verifier in the future. 😄

**Pull request merged**